### PR TITLE
Limelight note detection

### PR DESCRIPTION
**Pull request by 1076software**
Switching from USB Camera to Limelight for note detection, new driver controls for note pickup

### CODE REVIEW

File: notedetector.py [hunk 1]
```diff
@@ -1,92 +1,54 @@
...
-        self.lastY = self.getYMax
+    def getTargetErrorX(self):
+        return self.getTargetErrorY() * math.tan(math.radians(self.getTargetAngleX())) + self.config['CAMERA_OFFSET_X']
 
-        if self.sameCounterBBox > 10:    
-            return False
-        else:
-            return True
+    def getTargetErrorY(self):
+        return ((self.config['CAMERA_HEIGHT'] - self.config['NOTE_HEIGHT'])/math.tan(math.radians(self.getTargetAngleZ())))*(-1) + self.config['CAMERA_OFFSET_Y']
```
Comment (abdelazer) [inline notedetector.py:36]: I'd suggest breaking this expression into a set of named variables for clarity and then manipulate the variables.

File: notedetector.py [hunk 2]
```diff
@@ -1,92 +1,54 @@
...
-    def getTargetErrorY(self):
-        # y distance to the note
-        self.targetPixelZ = (self.config['CAMERA_PIXELS_Z'] - (self.getYMax())) - self.config['CAMERA_PIXELS_Z']/2
-        self.targetAngleZ = (self.targetPixelZ/(self.config['CAMERA_PIXELS_Z']/2))*(self.config['CAMERA_FOV_Z']/2) + self.config['CAMERA_ANGLE_ABOVE_HORIZONTAL']
-        self.targetErrorY = ((self.config['CAMERA_HEIGHT'] - self.config['NOTE_HEIGHT'])/math.tan(math.radians(self.targetAngleZ)))*(-1) + self.config['CAMERA_OFFSET_Y'] + self.config['NOTE_DIAMETER']/2
-        return self.targetErrorY  
\ No newline at end of file
+        self.lastHeartbeat = self.getHeartbeat()
+
+        if self.sameCounter > 10:
```
Comment (abdelazer) [inline notedetector.py:51]: Maybe this could be simplified to `return self.sameCounter < 11`?

File: robot.py [hunk 3]
```diff
@@ -245,7 +238,7 @@ def initSwervometer(self, config):
         return swervometer
 
     def initVision(self, config):
-        vision = Vision(NetworkTables.getTable('limelight'),
+        vision = Vision(NetworkTables.getTable('limelight-g'),
```
Comment (abdelazer) [inline robot.py:241]: It'd suggest putting limelight-g and limelight-note into robotconfig so they don't appear as magic values elsewhere.

File: robot.py [hunk 4]
```diff
@@ -588,10 +581,10 @@ def teleopDrivetrain(self):
             self.drivetrain.setWheelLock(False)
 
         if (driver.getYButton()):
-            # drive so that the center of the bot is on the center of the note, without rotating
-            self.drivetrain.alignWithNote(0, 0, None)
-            return False
-
+            self.drivetrain.alignWithNote(None, None, 0)
+            self.drivetrain.driveStraight(self.deadzoneCorrection(driver.getLeftY() * translational_clutch, self.driver.deadzone))
+            # self.drivetrain.alignWithNote(0, 0, None)
```
Comment (abdelazer) [inline robot.py:588]: Delete the commented-out line?

File: notedetector.py [hunk 5]
```diff
@@ -1,92 +1,54 @@
 from networktables import NetworkTables
 import math
 
+"""To Do:
+Make sure noteDrive_r PID is tuned
```
Comment (abdelazer) [inline notedetector.py:5]: Do we need to do this before merging?

**Comment by Cryplo** [comment]
I will hold on for merging now, we can plan to merge after we integrate auton with note detection again. I'll keep this PR open.

**Pull request merged**